### PR TITLE
Missing annotations added on applications module

### PR DESCRIPTION
**Pull request by marcosschroh**
Command: `mypy starlette/applications.py --disallow-untyped-defs --ignore-missing-imports --follow-imports=silent`

### CODE REVIEW

File: starlette/applications.py [hunk 1]
```diff
@@ -58,50 +60,55 @@ def debug(self) -> bool:
...
+    def websocket_route(self, path: str) -> typing.Callable:
+        def decorator(func: typing.Callable) -> typing.Callable:
             self.add_websocket_route(path, func)
             return func
 
         return decorator
 
     def __call__(self, scope: Scope) -> ASGIInstance:
-        scope["app"] = self
+        scope["app"] = self  # type: ignore
```
[the remark this answers is not in the document]
Comment (marcosschroh) [reply starlette/applications.py]: Yes, totally right. I have removed the comments!!

File: starlette/types.py [hunk 2]
```diff
@@ -9,3 +9,7 @@
 
 ASGIInstance = typing.Callable[[Receive, Send], typing.Awaitable[None]]
 ASGIApp = typing.Callable[[Scope], ASGIInstance]
+
+StrDict = typing.Mapping[str, str]
+BytesPairs = typing.List[typing.Tuple[bytes, bytes]]
+Methods = typing.Sequence[str]
```
[the remark this answers is not in the document]
Comment (marcosschroh) [reply starlette/types.py]: Yes, I had the same problem as before.I solved the annotations on a specific branch, I did a cherry pick to create several PR and I didn't not remove these direct notations :-(

File: starlette/applications.py [hunk 3]
```diff
@@ -58,45 +60,50 @@ def debug(self) -> bool:
...
+        route: typing.Callable,
+        methods: typing.Optional[Methods] = None,
+    ) -> None:
         if not inspect.isclass(route):
             route = request_response(route)
             if methods is None:
                 methods = ["GET"]
 
-        instance = Path(path, route, protocol="http", methods=methods)
+        instance = Path(path, route, protocol="http", methods=methods)  # type: ignore
```
[the remark this answers is not in the document]
Comment (marcosschroh) [reply starlette/applications.py]: Actually, on the signature of `__init__` method of `Path` class  methods is `typing.Sequence[str] = ()`, so we can do the same at for `add_route`: 
```python
def add_route(self, path: str, route: typing.Callable, methods: typing.Sequence[str] = ()) -> None:
    ...
    if not methods:
        methods = ("GET",)
```

What do you think?

File: starlette/types.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (lovelydinosaur) [reply starlette/types.py]: Don’t worry about cherry picking, just drop them from this PR in a commit would be fine. :)

**Comment by marcosschroh** [comment]
Merged with `master`. I've fixed some annotations as well.

File: starlette/applications.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (marcosschroh) [reply starlette/applications.py]: Great!!

File: starlette/applications.py [hunk 4]
```diff
@@ -1,15 +1,19 @@
...
 from starlette.routing import Path, PathPrefix, Router
 from starlette.types import ASGIApp, ASGIInstance, Receive, Scope, Send
 from starlette.websockets import WebSocket
-import asyncio
-import inspect
-import typing
 
 
-def request_response(func):
+Methods = typing.Sequence[str]
```
Comment (lovelydinosaur) [inline starlette/applications.py]: Okay, final thing. Let’s drop `Methods` and just use the `typing.Sequence[str]` annotation directly.

Then I think we’re done on this one!

**Review by lovelydinosaur — CHANGES_REQUESTED** [review]
Great stuff! One tiny bit left and then it’s 👍👍👍

File: starlette/applications.py (hunk 4 above)
Comment (marcosschroh) [reply starlette/applications.py]: OK!

**Review by lovelydinosaur — APPROVED** [review]

**Pull request merged**